CMakeLists.txt

```diff
@@ -3,6 +3,29 @@ cmake_minimum_required(VERSION 3.3)
 set (BASE_LLVM_VERSION 12.0.0)
 set(LLVM_SPIRV_VERSION ${BASE_LLVM_VERSION}.0)
 
+include(ExternalProject)
+
+# Download spirv.hpp from the official SPIRV-Headers repository.
+# One can skip this step in case of EXTERNAL_SPIRV_HEADERS_DIR path is set.
+if (NOT DEFINED EXTERNAL_SPIRV_HEADERS_DIR)
```

Comment:
I suggest that we re-use functionality provided by LLVM, i.e. we assume that SPIRV-Headers repo is an external project in terms of LLVM monorepo:

To check if it is available we use `LLVM_TOOL_SPIRV_HEADERS_BUILD` variable, which is automatically defined for us in  [llvm/CMakeLists.txt:127-131](https://github.com/llvm-project/llvm/blob/master/llvm/CMakeLists.txt#L127)

To get path to sources we use `LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR` variable, which is either supplied by user, or if `SPIRV-Headers` project was put into `llvm/projects` subdirectory, then it is automatically set by LLVM in  [llvm/cmake/modules/AddLLVM.cmake:1355-1348](https://github.com/llvm-project/llvm/blob/master/llvm/cmake/modules/AddLLVM.cmake#L1355)

Reply:
Done. I've ended up checking for `LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR` definition directly, because that variable can be configured manually to point to an external headers' installation without `LLVM_TOOL_SPIRV_HEADERS_BUILD` being set.